### ISSUE TITLE
Make the Charts Accessible

### ISSUE THREAD

**Issue by acouch** [cal-itp/reports#178]
**Background**

As a user, I want an accessible version of the charts produced in every report page so that I can understand the chart data using a screen reader or other accessibility tool.

**Acceptance criteria**

- [ ] All charts are published in a format that can be consumed by a screen reader
- [ ] A public standard like WCAG can be applied, if it is identified and applied

**Comment by acouch** [comment]
First step is choosing a charting library that doesn't just print out images. SVGs are ideal because the data can be exposed accessibly within the markup as opposed to canvas or images.

I've done some research on this. High Charts has the [most accessible](https://www.highcharts.com/docs/accessibility/accessibility-module) [options](https://www.highcharts.com/blog/accessibility/) I could find. However it is $150 a year license which I don't know if we can procure. Will find out.

I've reviewed the current D3-based, fully open-source charting libraries. Most of them are not actively maintained. I'm investigating [BillboardJS](https://github.com/naver/billboard.js) as option for a library that we can use that I can enhance.

I'm using a couple of D3-based guides https://www.a11ywithlindsey.com/blog/accessibility-d3-bar-charts, https://css-tricks.com/accessible-svgs, https://handbook.digital.ca.gov/tech/fast-dashboards; and will evaluate creating a solution that is just custom D3. I'd like to use a library as it will make it quicker in the future to add more types of graphs with less dev overhead.

Another consideration is that the current charts don't have a legend on them:

![image](https://user-images.githubusercontent.com/512243/187493828-8fd10dcb-6b26-464a-b2bd-1a3b01c0fa21.png)

Adding a legend might be a design consideration and take more resources on our end. Will discuss on that end as well.

**Comment by acouch** [comment]
This looks most possible with billboardjs. This comment shows a mechanism for ensuring either an accessible legend is added and / or each point of the chart has an accessible text element https://github.com/naver/billboard.js/issues/2245#issuecomment-896557994

**Comment by ryon** [comment]
I reviewed the articles and the billboard.js example. Using the latter I put together a small prototype of one of our charts along with `tabindex` and `aria-label` attributes on the individual data points: https://codepen.io/ryon/pen/NWMONBZ?editors=1010

This seems to work, and switching to SVG charts in general should be a win. I only wonder if this particular approach (grafting attributes onto billboard's output, as opposed to the more intentional/structured designs in some of the other links) offers enough information, or well-organized enough, to be useful to an actual screen reader user. It would be great if we could arrange some feedback from a consultant or real user if at all possible.

Another approach I've considered, but not sure if it's appropriate, would be to more or less hide the chart itself from screen readers as decorative, and instead offer a plain data `<table>` in the markup wherever we have a chart. The table would offer structured information in a way that is presumably accessible, familiar, and navigable to screen readers, though I hesitate to act on a presumption!

**Comment by acouch** [comment]
@ryon This looks good! I think we can move forward with tabbed example for now. Some feedback for when it is actually put on the site is to match the font of the existing site and to make sure the labels are as larger than the codepen example.
